src/tree/GroupTreeItemBase.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "@microsoft/vscode-azext-utils";
+import { localize } from "../utils/localize";
+import { ResolvableTreeItem } from "./ResolvableTreeItem";
+import { ShallowResourceTreeItem } from "./ShallowResourceTreeItem";
+
+export abstract class GroupTreeItemBase extends AzExtParentTreeItem {
+    public readonly childTypeLabel: string = localize('resource', 'Resource');
+    public treeMap: { [key: string]: (ResolvableTreeItem | ShallowResourceTreeItem) } = {};
+    public abstract label;
+
+    public readonly cTime: number = Date.now();
```

Comment:
Are these really required now?!

Reply:
It's used here, but I'm unsure what purpose `cTime` actually serves.

https://github.com/microsoft/vscode-azureresourcegroups/blob/d38b07978ba0c41516f8cf9f9e4eb5d35528ca55/src/commands/tags/TagFileSystem.ts#L31

Reply:
Yeah, I think it was mostly to see if the tags needed to be updated by comparing the times, but as per this comment we can probably expand its capabilities:

```
/**
 * For now this file system only supports editing tags.
 * However, the scheme was left generic so that it could support editing other stuff in this extension without needing to create a whole new file system
 */
```